students/21th/hyunchanpark/main.js

```diff
@@ -0,0 +1,95 @@
+const commentForm = document.querySelector(".comment-form");
+
+const commentInput = document.querySelector(".comment-input");
+
+const otherComments = document.querySelector(".other-comments");
+
+const postButton = document.querySelector(".post-button");
+
+let heartButtons = document.querySelectorAll(".fa-heart");
+
+let removeButtons = document.querySelectorAll("fa-trash-alt");
+
+const isDisabledButton = () => {
+  let isAbleButton = false;
+  if (commentInput.value.length !== 0) {
+    postButton.disabled = false;
+    isAbleButton = true;
+  } else {
+    postButton.disabled = true;
+    isAbleButton = false;
+  }
+
+  return isAbleButton;
+};
+
+const toggleHeart = (node) => {
+  node.addEventListener("click", () => {
+    if (node.classList.contains("fas")) {
+      node.classList.add("far");
+      node.classList.remove("fas");
+    } else if (node.classList.contains("far")) {
+      node.classList.add("fas");
+      node.classList.remove("far");
+    }
```

Comment:
- 전체를 반복해서 작성하기 보단, 아래와 같이 변화하는 부분에 대해서만 조건을 걸어 바꿔주시는 방식으로 코드를 작성하는게 유지보수 측면에서 더 좋은 코딩 습관이 될 것 같습니다! 참고해보세요~
```suggestion
    const classToAdd = node.classList.contains('fas') ? 'far' : 'fas';
    const classToRemove = node.classList.contains('fas') ? 'fas' : 'far';
    node.classList.add(classToAdd);
    node.classList.remove(classToRemove);
```